lib/std/os.zig

```diff
@@ -5012,11 +5012,10 @@ pub fn sigaltstack(ss: ?*stack_t, old_ss: ?*stack_t) SigaltstackError!void {
 }
 
 /// Examine and change a signal action.
-pub fn sigaction(sig: u6, act: ?*const Sigaction, oact: ?*Sigaction) void {
+pub fn sigaction(sig: u6, noalias act: ?*const Sigaction, noalias oact: ?*Sigaction) error{OperationNotSupported}!void {
     switch (errno(system.sigaction(sig, act, oact))) {
         .SUCCESS => return,
-        .FAULT => unreachable,
```

Comment:
I removed the `.FAULT` case (for now) completely, as I don't know why was there in the first place given that the `else` branch also satisfies it. There may be some deeper reason behind it, but idk.

`INVAL` and `NOSYS` should catch most systems with unexpected behavoir without looping. But it might be appropriate to return `Unexpected` instead of using `unreachable`.